mne/viz/evoked.py

```diff
@@ -311,10 +311,18 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
                         for i in bad_ch_idx:
                             if i in idx:
                                 colors[idx.index(i)] = 'r'
-                    for ch_idx in range(len(D)):
+
+                    # find the channels with the least activity,
+                    # and map them in front of the more active ones
+                    auc = np.trapz((np.abs(D - D.mean(1).reshape(-1, 1))))
```

Comment:
It maybe worth a parameter. In principle I like this, I know these kinds
kind of situations where the order of plotting can make a difference.
zorder could take a function that is used for sorting. Same for
transparency.

On Tue, May 3, 2016 at 6:08 PM jona-sassenhagen <notifications@github.com>
wrote:

> In mne/viz/evoked.py
> <https://github.com/mne-tools/mne-python/pull/3199#discussion_r61908804>:
>
> > @@ -311,10 +311,18 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
> >                          for i in bad_ch_idx:
> >                              if i in idx:
> >                                  colors[idx.index(i)] = 'r'
> > -                    for ch_idx in range(len(D)):
> > +
> > +                    # find the channels with the least activity,
> > +                    # and map them in front of the more active ones
> > +                    auc = np.trapz((np.abs(D - D.mean(1).reshape(-1, 1))))
>
> Let's first decide on this actually being an improvement at all before
> deciding on the implementation, I would think ...
> It works well with the standard deviation, so much I can say at least.
>
> I think we possibly should make SOME principled decision about how to
> z-order these, though.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3199/files/c39ebf71608334a65c0678a0354f8ee2697c6645#r61908804>
>